fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java

```diff
@@ -0,0 +1,54 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.common.proc;
+
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.util.TimeUtils;
+import com.starrocks.lake.compaction.CompactionContext;
+import com.starrocks.lake.compaction.CompactionManager;
+import com.starrocks.lake.compaction.PartitionIdentifier;
+import com.starrocks.server.GlobalStateMgr;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CompactionsProcDir implements ProcDirInterface {
+    private List<String> titles = new ArrayList<>();
+
+    public CompactionsProcDir() {
+        titles.add("Partition");
+        titles.add("TxnID");
+        titles.add("StartTime");
+    }
+
+    @Override
+    public boolean register(String name, ProcNodeInterface node) {
+        return false;
+    }
+
+    @Override
+    public ProcNodeInterface lookup(String name) throws AnalysisException {
+        return null;
+    }
+
+    @Override
+    public ProcResult fetchResult() throws AnalysisException {
+        BaseProcResult result = new BaseProcResult();
+        result.setNames(titles);
+        CompactionManager compactionManager = GlobalStateMgr.getCurrentState().getCompactionManager();
+        if (compactionManager == null) {
+            return result;
+        }
+        ConcurrentHashMap<PartitionIdentifier, CompactionContext> runningCompactions = compactionManager.getRunningCompactions();
+        for (CompactionContext context : runningCompactions.values()) {
+            List<String> row = new ArrayList<>();
+            row.add(String.valueOf(context.getPartitionName()));
```

Comment:
Better to add table id or table name.

Reply:
`context.getPartitionName()` contains database and table name.